infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/LogReplicationClientRouter.java

```diff
@@ -144,11 +144,15 @@ public IClientRouter addClient(IClient client) {
         client.setRouter(this);
 
         // Iterate through all types of CorfuMsgType, registering the handler
-        client.getHandledTypes().stream()
-                .forEach(x -> {
-                    handlerMap.put(x, client);
-                    log.info("Registered client to handle messages of type {}", x);
-                });
+        try {
+            client.getHandledTypes().stream()
+                    .forEach(x -> {
+                        handlerMap.put(x, client);
+                        log.info("Registered client to handle messages of type {}", x);
+                    });
+        } catch (UnsupportedOperationException ex) {
+            log.warn("No registered CorfuMsg handler for client {}", client);
```

Comment:
log.error() with the exception too? This might be one of the things we would need to handle for rolling upgrade in the future, if a client message arrives that does not have any registered handler right, so perhaps a comment or TODO for the same?

Reply:
I have adjusted the logging as suggested, but I want to point out that this is for legacy `CorfuMsg` handlers. The default behavior of these legacy methods is to throw an exception when invoked but not implemented, and will be cleaned up in the last PR once all RPCs are done over Protobuf.

Reply:
sounds good